src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -171,6 +171,10 @@ else if (!hasText(leftCurly)) {
                     ast.getText());
             }
         }
+
+        if (ast.getType() == TokenTypes.LITERAL_CASE) {
+            checkForEmptyCase(ast);
+        }
```

Comment:
@romani  
I am sorry I thought you asked why did I use a separate method. I processed it separately since to check for empty case the `slist` token which is examined is the next sibling of `LITERAL_CASE` and hence `findFirstToken` can't be used there directly ! Also since we are to log for empty `case`, `leftCurly` in the `visitToken` method would have to be made equal to the argument supplied i.e `ast`. Hence there has to be some amount of separate processing involved I feel and it also helps avoid confusion imho. (In other cases the `slist` token which should be examined for empty block are generally a child of the argument `ast` supplied but when the `ast` argument is of the type `LITERAL_CASE` then the slist which is to be examined is a sibling and hence one would to either use `ast.getNextSibbling()` or `ast.getParent().findFirstToken(TokenTypes.SLIST)` and hence the need for separate processing).